psqlextra/locking.py

```diff
@@ -0,0 +1,97 @@
+from enum import Enum
+from typing import Optional, Type
+
+from django.db import DEFAULT_DB_ALIAS, connections, models
+
+
+class PostgresTableLockMode(Enum):
+    """List of table locking modes.
+
+    See: https://www.postgresql.org/docs/current/explicit-locking.html
+    """
+
+    ACCESS_SHARE = "ACCESS SHARE"
+    ROW_SHARE = "ROW SHARE"
+    ROW_EXCLUSIVE = "ROW EXCLUSIVE"
+    SHARE_UPDATE_EXCLUSIVE = "SHARE UPDATE EXCLUSIVE"
+    SHARE = "SHARE"
+    SHARE_ROW_EXCLUSIVE = "SHARE ROW EXCLUSIVE"
+    EXCLUSIVE = "EXCLUSIVE"
+    ACCESS_EXCLUSIVE = "ACCESS EXCLUSIVE"
+
+
+def postgres_lock_table(
+    table_name: str,
+    lock_mode: PostgresTableLockMode,
+    *,
+    schema_name: Optional[str] = None,
+    using: str = DEFAULT_DB_ALIAS,
+) -> None:
+    """Locks the specified table with the specified mode.
+
+    The lock is held until the end of the current transaction.
+
+    Arguments:
+        table_name:
+            Unquoted table name to acquire the lock on.
+
+        lock_mode:
+            Type of lock to acquire.
+
+        schema_name:
+            Optionally, the unquoted name of the schema
+            the table to lock is in. If not specified,
+            the table name is resolved by PostgreSQL
+            using it's ``search_path``.
+
+        using:
+            Name of the database alias to use.
```

Comment:
```suggestion
            Optional name of the database alias to use.
```

Reply:
I've addressed this by making the description for this parameter across the code base consistent:

> Optional name of the database connection to use.